vllm/logger.py

```diff
@@ -1,61 +1,255 @@
+        if unstructured_filepath:
+            # Unstructured file logging configuration
+            logger.add(
+                str(unstructured_filepath),
+                rotation=rotation,
+                retention=retention,
+                enqueue=True,
+                backtrace=True,
+                level=level.upper(),
+                serialize=False,
+                format=format,
+            )
+        if structured_filepath:
+            # Structured file logging configuration
+            logger.add(
+                str(structured_filepath),
+                rotation=rotation,
+                retention=retention,
+                enqueue=True,
+                backtrace=True,
+                level=level.upper(),
+                serialize=False,
+                format=cls.formatter,
+            )
 
-    def __init__(self, fmt, datefmt=None):
-        logging.Formatter.__init__(self, fmt, datefmt)
+        # Basic configuration for intercepting standard logging messages
+        logging.basicConfig(handlers=[InterceptHandler()], level=0)
 
-    def format(self, record):
-        msg = logging.Formatter.format(self, record)
-        if record.message != "":
-            parts = msg.split(record.message)
-            msg = msg.replace("\n", "\r\n" + parts[0])
-        return msg
+        return logger.bind(name="vllm")
 
+    @classmethod
+    def load_logging_config(cls, config_path: Path):
+        """Load logging configuration from a JSON file.
 
-_root_logger = logging.getLogger("vllm")
-_default_handler = None
+        Args:
+            config_path (Path): Path to the configuration file.
 
+        Returns:
+            dict: Loaded configuration.
+        """
+        config = None
+        with open(config_path) as config_file:
+            config = json.load(config_file)
+        return config
 
-def _setup_logger():
-    _root_logger.setLevel(logging.DEBUG)
-    global _default_handler
-    if _default_handler is None:
-        _default_handler = logging.StreamHandler(sys.stdout)
-        _default_handler.flush = sys.stdout.flush  # type: ignore
-        _default_handler.setLevel(logging.INFO)
-        _root_logger.addHandler(_default_handler)
-    fmt = NewLineFormatter(_FORMAT, datefmt=_DATE_FORMAT)
-    _default_handler.setFormatter(fmt)
-    # Setting this will avoid the message
-    # being propagated to the parent logger.
-    _root_logger.propagate = False
 
+dir_path = os.path.dirname(os.path.realpath(__file__))
+default_config_path = f"{dir_path}/default_logging_config.json"
+config_path = Path(os.getenv("VLLM_LOGGING_CONFIG_PATH", default_config_path))
+_root_logger = None
 
-# The logger is initialized when the module is imported.
-# This is thread-safe as the module is only imported once,
-# guaranteed by the Python GIL.
 if VLLM_CONFIGURE_LOGGING:
-    _setup_logger()
+    _root_logger = CustomizeLogger.make_logger(config_path)
+else:
+    _root_logger = logging.getLogger("vllm")
```

Comment:
IIUC, if VLLM_CONFIGURE_LOGGING is False, doesn't this have regression? I.e., don't we need to do 

```
def _setup_logger():
    _root_logger.setLevel(logging.DEBUG)
    global _default_handler
    if _default_handler is None:
        _default_handler = logging.StreamHandler(sys.stdout)
        _default_handler.flush = sys.stdout.flush  # type: ignore
        _default_handler.setLevel(logging.INFO)
        _root_logger.addHandler(_default_handler)
    fmt = NewLineFormatter(_FORMAT, datefmt=_DATE_FORMAT)
    _default_handler.setFormatter(fmt)
    # Setting this will avoid the message
    # being propagated to the parent logger.
    _root_logger.propagate = False
```
?